### PR TITLE
[ADD] odoo-addons-relative-import

### PR DESCRIPTION
**Pull request by moylop260**
Inspired from https://github.com/Vauxoo/stock-logistics-workflow/pull/16#discussion_r71438218

### CODE REVIEW

**Comment by dreispt** [comment]
:+1:

**Comment by max3903** [comment]
:+1:

**Comment by pedrobaeza** [comment]
I don't get what is the problem you want to address.

**Comment by moylop260** [comment]
This check catch the following case:

- `addons/my_module/tests/test_demo.py` with the following content:
 - `from openerp.addons.my_module import something`

But if you are `my_module` then you don't need import with `from openerp.addons.my_module...` because you are in we should import with `from .. my_module import something`

It's a variant of a relative-import but using the dynamic sys.modules of odoo.

File: pylint_odoo/checkers/modules_odoo.py [hunk 1]
```diff
@@ -77,6 +78,12 @@
         'wrong-tabs-instead-of-spaces',
         settings.DESC_DFLT
     ),
+    'W%d50' % settings.BASE_OMODULE_ID: (
+        'Odoo addons relative import, should use "." '
```
Comment (pedrobaeza) [inline pylint_odoo/checkers/modules_odoo.py]: A more explanatory text should be: "Same Odoo module relative import. You should use..."

Comment (pedrobaeza) [inline pylint_odoo/checkers/modules_odoo.py]: Sorry, it should be "Same Odoo module absolute import. You should use..."

Comment (moylop260) [inline pylint_odoo/checkers/modules_odoo.py]: Thanks for feedback
Change applied!

**Comment by pedrobaeza** [comment]
Can you check the Travis error?

**Comment by dreispt** [comment]
@moylop260 AFAICT for controllers we can't use this relative import

**Comment by moylop260** [comment]
@dreispt 
Thanks for feedback.
I don't understand you
Could you send us a example?

**Comment by dreispt** [comment]
When you want to import from the inside the same module, relative imports are OK.
But what about when you want to import from other modules?
Odoo always uses an abolute path, and this is commonly used in controllers.
Example: https://github.com/odoo/odoo/blob/9.0/addons/website_sale/controllers/main.py#L10

**Comment by moylop260** [comment]
That example is a good case and not is emitted a message there because: The module is `website_sale` and you are importing `website`

The wrong case is if the module is `website_sale` and you are importing `website_sale`

**Comment by dreispt** [comment]
Ok then.
Note that CI is red.

**Comment by moylop260** [comment]
FYI error unrelated
FIX https://github.com/OCA/pylint-odoo/pull/61

**Comment by pedrobaeza** [comment]
Please rebase

**Comment by moylop260** [comment]
rebased
Thanks @pedrobaeza

**Comment by pedrobaeza** [comment]
:+1:

**Pull request merged**